File: apps/api/src/Tymblok.Core/Entities/Category.cs
namespace Tymblok.Core.Entities;

public class Category : BaseEntity
{
    public Guid? UserId { get; set; } // Null for system categories
    public ApplicationUser? User { get; set; }

    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = "#6366f1"; // Hex color
    public string Icon { get; set; } = "default"; // Icon identifier

    public bool IsSystem { get; set; } = false; // System categories can't be deleted

    // Navigation
    public ICollection<TimeBlock> TimeBlocks { get; set; } = [];
}

File: packages/api-client/src/types/category.ts
export interface CategoryDto {
  id: string;
  name: string;
  color: string;
  icon: string;
  isSystem: boolean;
  createdAt: string;
}

export interface CreateCategoryRequest {
  name: string;
  color: string;
  icon: string;
}

export interface UpdateCategoryRequest {
  name: string;
  color: string;
  icon: string;
}

export interface CategoriesResponse {
  categories: CategoryDto[];
}
